sqlglot/expressions.py

```diff
@@ -2556,9 +2556,20 @@ def with_(
     "sample": False,
     "settings": False,
     "format": False,
+    "option": False,
 }
 
 
+# https://learn.microsoft.com/en-us/sql/t-sql/queries/option-clause-transact-sql?view=sql-server-ver16
+# https://learn.microsoft.com/en-us/sql/t-sql/queries/hints-transact-sql-query?view=sql-server-ver16
+class Option(Expression):
+    arg_types = {"option_type": True, "option_value": False, "requires_equals": False}
```

Comment:
Maybe rename this to something like `QueryOption`? If possible, let's try not to use a generic name like "option" for something that's only relevant to T-SQL.

Reply:
Also, the `option_` prefix is redundant - we already know this is under `Option`.

Maybe use something like `this` and `expression` for the first two args? The `requires_equals` arg can be omitted if you make a map with known options that need the `=` which you can leverage at generation time.

Reply:
Done